### PR TITLE
lowering and codegen of COMMON blocks

### PR DESCRIPTION
**Pull request by schweitzpgi**

### CODE REVIEW

File: flang/lib/Lower/Bridge.cpp [hunk 1]
```diff
@@ -789,12 +789,11 @@ class FirConverter : public Fortran::lower::AbstractConverter {
...
-                         ? genFIRLoopIndex(*info.stepExpr, type)
-                         : info.isStructured()
-                               ? builder->create<mlir::ConstantIndexOp>(loc, 1)
-                               : builder->createIntegerConstant(
-                                     loc, info.loopVariableType, 1);
+    info.stepValue =
+        info.stepExpr.has_value() ? genFIRLoopIndex(*info.stepExpr, type)
+        : info.isStructured()
+            ? builder->create<mlir::ConstantIndexOp>(loc, 1)
+            : builder->createIntegerConstant(loc, info.loopVariableType, 1);
```
Comment (vdonaldson) [inline flang/lib/Lower/Bridge.cpp:796]: We need to do something about this yo-yo diff.  Whichever instance is correct, let's stop changing it.

Comment (schweitzpgi) [reply flang/lib/Lower/Bridge.cpp:796]: Agreed. I'll add that.

File: flang/lib/Lower/CMakeLists.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-error -Wno-unused-parameter")
```
Comment (Sameeranjoshi) [inline flang/lib/Lower/CMakeLists.txt:1]: Any reason to do this change :)? I see failures during build #263

Comment (schweitzpgi) [reply flang/lib/Lower/CMakeLists.txt:1]: Absolutely. It's time to fix all the warnings.

**Pull request merged**